### ISSUE TITLE
PyPI Filter fails to retrieve packaging info

### ISSUE THREAD

**Issue by MagielBruntink** [fasten-project/pypi-tools#12]
PyPI Filter appears broken at the moment. It cannot get the packaging information it needs from its input coordinate and/or PyPI.

This happens for all the coordinates I have tried. One example with DC is:
http://localhost:9080/api/pypi/packages/pydriller/2.4

The following message appears on `fasten.PyPI.releases.ingest` as normal:
```
{
   "ingested": true,
   "project": {
      "urls": [
         {
            "yanked": false,
            "yanked_reason": null,
            "comment_text": "",
            "has_sig": false,
            "md5_digest": "4d2fbf28b0893f3e516acd32052142d3",
            "url": "https://files.pythonhosted.org/packages/87/f2/5609ea23729e7c32d8a8aad475a184ce14eb8dc7240dbc42cd46aba506e3/PyDriller-2.4-py3-none-any.whl",
            "upload_time": "2023-01-21T13:58:38",
            "filename": "PyDriller-2.4-py3-none-any.whl",
            "upload_time_iso_8601": "2023-01-21T13:58:38.117467Z",
            "size": 33279,
            "downloads": -1,
            "python_version": "py3",
            "digests": {
               "sha256": "2663dff95cd592cabdc624c49a407ca3ef0f83fb81d6023f2271418c521faea7",
               "blake2b_256": "87f25609ea23729e7c32d8a8aad475a184ce14eb8dc7240dbc42cd46aba506e3",
               "md5": "4d2fbf28b0893f3e516acd32052142d3"
            },
            "packagetype": "bdist_wheel",
            "requires_python": ">=3.5"
         }
      ],
      "vulnerabilities": [],
      "last_serial": 16512458,
      "info": {
         "keywords": "",
         "package_url": "https://pypi.org/project/PyDriller/",
         "description": "PyDriller is a Python framework that helps developers on mining software repositories. With PyDriller you can easily extract information from any Git repository, such as commits, developers, modifications, diffs, and source codes, and quickly export CSV files.\n\n\n",
         "project_url": "https://pypi.org/project/PyDriller/",
         "project_urls": {
            "Homepage": "https://github.com/ishepard/pydriller"
         },
         "release_url": "https://pypi.org/project/PyDriller/2.4/",
         "maintainer": "",
         "platform": null,
         "home_page": "https://github.com/ishepard/pydriller",
         "docs_url": null,
         "requires_dist": [
            "gitpython",
            "pytz",
            "types-pytz",
            "lizard"
         ],
         "downloads": {
            "last_month": -1,
            "last_day": -1,
            "last_week": -1
         },
         "classifiers": [
            "Development Status :: 5 - Production/Stable",
            "Environment :: Console",
            "Intended Audience :: Developers",
            "Intended Audience :: Science/Research",
            "License :: OSI Approved :: Apache Software License",
            "Operating System :: MacOS :: MacOS X",
            "Operating System :: Microsoft :: Windows",
            "Operating System :: OS Independent",
            "Operating System :: POSIX",
            "Programming Language :: Python :: 3.6",
            "Programming Language :: Python :: 3.7",
            "Programming Language :: Python :: 3.8",
            "Programming Language :: Python :: 3.9",
            "Topic :: Software Development :: Libraries :: Python Modules"
         ],
         "download_url": "",
         "author_email": "spadini.davide@gmail.com",
         "requires_python": ">=3.5",
         "bugtrack_url": null,
         "summary": "Framework for MSR",
         "yanked": false,
         "yanked_reason": null,
         "author": "Davide Spadini",
         "version": "2.4",
         "license": "Apache License",
         "name": "PyDriller",
         "maintainer_email": "",
         "description_content_type": ""
      }
   },
   "title": "pydriller 2.4"
}
```

Out error message in PyPI Filter container log:
```
Could not retrieve packaging info from {
  "ingested": true,
  "project": {
    "urls": [
      {
        "yanked": false,
        "yanked_reason": null,
        "comment_text": "",
        "has_sig": false,
        "md5_digest": "4d2fbf28b0893f3e516acd32052142d3",
        "url": "https://files.pythonhosted.org/packages/87/f2/5609ea23729e7c32d8a8aad475a184ce14eb8dc7240dbc42cd46aba506e3/PyDriller-2.4-py3-none-any.whl",
        "upload_time": "2023-01-21T13:58:38",
        "filename": "PyDriller-2.4-py3-none-any.whl",
        "upload_time_iso_8601": "2023-01-21T13:58:38.117467Z",
        "size": 33279,
        "downloads": -1,
        "python_version": "py3",
        "digests": {
          "sha256": "2663dff95cd592cabdc624c49a407ca3ef0f83fb81d6023f2271418c521faea7",
          "blake2b_256": "87f25609ea23729e7c32d8a8aad475a184ce14eb8dc7240dbc42cd46aba506e3",
          "md5": "4d2fbf28b0893f3e516acd32052142d3"
        },
        "packagetype": "bdist_wheel",
        "requires_python": ">=3.5"
      }
    ],
    "vulnerabilities": [],
    "last_serial": 16512458,
    "info": {
      "keywords": "",
      "package_url": "https://pypi.org/project/PyDriller/",
      "description": "PyDriller is a Python framework that helps developers on mining software repositories. With PyDriller you can easily extract information from any Git repository, such as commits, developers, modifications, diffs, and source codes, and quickly export CSV files.\n\n\n",
      "project_url": "https://pypi.org/project/PyDriller/",
      "project_urls": {
        "Homepage": "https://github.com/ishepard/pydriller"
      },
      "release_url": "https://pypi.org/project/PyDriller/2.4/",
      "maintainer": "",
      "platform": null,
      "home_page": "https://github.com/ishepard/pydriller",
      "docs_url": null,
      "requires_dist": [
        "gitpython",
        "pytz",
        "types-pytz",
        "lizard"
      ],
      "downloads": {
        "last_month": -1,
        "last_day": -1,
        "last_week": -1
      },
      "classifiers": [
        "Development Status :: 5 - Production/Stable",
        "Environment :: Console",
        "Intended Audience :: Developers",
        "Intended Audience :: Science/Research",
        "License :: OSI Approved :: Apache Software License",
        "Operating System :: MacOS :: MacOS X",
        "Operating System :: Microsoft :: Windows",
        "Operating System :: OS Independent",
        "Operating System :: POSIX",
        "Programming Language :: Python :: 3.6",
        "Programming Language :: Python :: 3.7",
        "Programming Language :: Python :: 3.8",
        "Programming Language :: Python :: 3.9",
        "Topic :: Software Development :: Libraries :: Python Modules"
      ],
      "download_url": "",
      "author_email": "spadini.davide@gmail.com",
      "requires_python": ">=3.5",
      "bugtrack_url": null,
      "summary": "Framework for MSR",
      "yanked": false,
      "yanked_reason": null,
      "author": "Davide Spadini",
      "version": "2.4",
      "license": "Apache License",
      "name": "PyDriller",
      "maintainer_email": "",
      "description_content_type": ""
    }
  },
  "title": "pydriller 2.4"
}
```

**Comment by gdrosos** [comment]
Thanks @MagielBruntink, I think this issue is similar to https://github.com/fasten-project/fasten/issues/488
I have submited a [Pull Request](https://github.com/fasten-project/fasten/pull/502)  trying to address the issue. Before merging the PR we discussed changing the message `Package version not found` because it was misleading. But I think this issue is more complicated than it seems: 

There are cases where we want to display the aforementioned message (when the package version does not exist on PyPI or Maven and null is returned from the database). The issue lies in the of use the `ingestArtifactIfNecessary` method which is true when an ingestion is needed and false otherwise. The problem is that in 2 different cases  the method yields the same result. Precisely, when a package version is not found on pypi, and when a package is already ingested but not yet on the db, the method will return false since in both cases an ingestion is not necessary. This leads to the generic `Package version not found` message. The second case does not occur to Maven as the coordinate becomes ingested at later stages in the pipeline.

In order to tackle this issue we need to change the logic behind the generic & PyPI-specific ingestion methods, but I think this should be done in a Pull Request separatelly.
